pkg/metrics/metrics.go

```diff
@@ -256,25 +215,60 @@ func isEndpointsValid(nodes *v1.NodeList, endpoints *v1.Endpoints) bool {
 	return true
 }
 
-// updateServiceMonitors patches the metrics Service Monitor to include required fields to display node graphs on the
-// OpenShift console. Console graph queries require metrics endpoint target name to be node name, however
-// windows_exporter returns node IP. We replace the target name by adding `replace` action field to the ServiceMonitor
-// object that replaces node IP to node name as the metrics endpoint target.
-func updateServiceMonitors(cfg *rest.Config, namespace string) error {
-
-	patchData := fmt.Sprintf("[{\"op\": \"replace\", \"path\": \"/spec/endpoints/0\", "+
-		"\"value\":{\"path\": \"/%s\",\"port\": \"%s\",\"relabelings\": [{\"action\": \"replace\", \"regex\": \"(.*)\", "+
-		"\"replacement\": \"$1\", \"sourceLabels\": [\"__meta_kubernetes_endpoint_address_target_name\"],"+
-		"\"targetLabel\": \"instance\"}]}}]", PortName, PortName)
+// Configure takes care of all the required configuration steps
+// for Prometheus monitoring like validating monitoring label
+// and creating metrics Endpoints object.
+func (c *Config) Configure(ctx context.Context) error {
+	// validate if cluster monitoring is enabled in the operator namespace
+	if err := c.validate(ctx); err != nil {
+		log.Error(err, "error validating cluster monitoring label")
+		return nil
+	}
+	// Create Metrics Endpoint object in the operator namespace
+	if err := c.createEndpoint(); err != nil {
+		return errors.Wrap(err, "error creating metrics Endpoint")
+	}
+	return nil
+}
 
-	mclient, err := monclient.NewForConfig(cfg)
+// validate will verify if cluster monitoring is enabled in the operator namespace.
+// If the label is not present, it will log and send warning events to the user.
+func (c *Config) validate(ctx context.Context) error {
+	// validate if metrics label is added to namespace
+	wmcoNamespace, err := c.CoreV1().Namespaces().Get(ctx, c.namespace, metav1.GetOptions{})
 	if err != nil {
-		return errors.Wrap(err, "error creating monitoring client")
+		return errors.Wrap(err, "error getting operator namespace")
+	}
+	if wmcoNamespace.Labels["openshift.io/cluster-monitoring"] != "true" {
+		metricsEnabled = false
+		c.Recorder.Eventf(wmcoNamespace, core.EventTypeWarning, "labelValidationFailed",
+			"Cluster monitoring openshift.io/cluster-monitoring label is not enabled in %s namespace", c.namespace)
+		return errors.Errorf("monitoring not enabled in %s namespace", c.namespace)
```

Comment:
Why is this an error? It seems the function is more about checking if metrics is enabled or not and should return `true` or `false` in addition to an error.

Reply:
can be implemented as you suggested, the current implementation looks for an error for returning

Reply:
Please see previous comment.